Add Ansible

KeyboardInterrupt opened the pull request:
<!-- Congrats on creating an Awesome list! 🎉 -->

<!-- Please fill in the below placeholders -->

https://github.com/KeyboardInterrupt/awesome-ansible

Ansible is a Configuration Management System and People have built Awesome Tools and Automation around and with it.
The goal of the list is to create a curated List of great resources around Ansible and it's Ecosystem.

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **Don't waste my time.** Do a good job, adhere to all the guidelines, and be responsive.
- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** --> #1838 & #1794 & #1816 
	Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration. **Just commenting “looks good” or simply marking the pull request as approved does not count!** You have to actually point out mistakes or improvement suggestions.
- You have read and understood the [instructions for creating a list](create-list.md).
- This pull request has a title in the format `Add Name of List`.
	- ✅ `Add Swift`
	- ✅ `Add Software Architecture`
	- ❌ `Update readme.md`
	- ❌ `Add Awesome Swift`
	- ❌ `Add swift`
	- ❌ `Adding Swift`
	- ❌ `Added Swift`
- Your entry here should include a short description about the project/theme of the list. **It should not describe the list itself.** The first character should be uppercase and the description should end in a dot. It should be an objective description and not a tagline or marketing blurb.
	- ✅ `- [iOS](…) - Mobile operating system for Apple phones and tablets.`
	- ✅ `- [Framer](…) - Prototyping interactive UI designs.`
	- ❌ `- [iOS](…) - Resources and tools for iOS development.`
	- ❌ `- [Framer](…)`
	- ❌ `- [Framer](…) - prototyping interactive UI designs`
- Your entry should be added at the bottom of the appropriate category.
- The suggested Awesome list complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- Don't open a Draft / WIP pull request while you work on the guidelines. A pull request should be 100% ready and should adhere to all the guidelines when you open it.
- Run [`awesome-lint`](https://github.com/sindresorhus/awesome-lint) on your list and fix the reported issues. If there are false-positives or things that cannot/shouldn't be fixed, please [report it](https://github.com/sindresorhus/awesome-lint/issues/new).
- The default branch should be named [`main`, not `master`](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/).
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- ✅ `Mobile operating system for Apple phones and tablets.`
	- ✅ `Prototyping interactive UI designs.`
	- ❌ `Resources and tools for iOS development.`
	- ❌ `Awesome Framer packages and tools.`
- It's the result of hard work and the best I could possibly produce.
	**If you have not put in considerable effort into your list, your pull request will be immediately closed.**
- The repo name of your list should be in lowercase slug format: `awesome-name-of-list`.
	- ✅ `awesome-swift`
	- ✅ `awesome-web-typography`
	- ❌ `awesome-Swift`
	- ❌ `AwesomeWebTypography`
- The heading title of your list should be in [title case](https://capitalizemytitle.com/) format: `# Awesome Name of List`.
	- ✅ `# Awesome Swift`
	- ✅ `# Awesome Web Typography`
	- ❌ `# awesome-swift`
	- ❌ `# AwesomeSwift`
- Non-generated Markdown file in a GitHub repo.
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate. Please search for existing submissions.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Does not contain items that are unmaintained, has archived repo, deprecated, or missing docs. If you really need to include such items, they should be in a separate Markdown file.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- **The image should be high-DPI.** Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an appropriate license.
	- **We strongly recommend the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), but any [Creative Commons license](https://creativecommons.org/choose/) will work.**
		- Tip: You can quickly add it to your repo by going to this URL: `https://github.com/<user>/<repo>/community/license/new?branch=master&template=cc0-1.0` (replace `<user>` and `<repo>` accordingly).
	- A code license like MIT, BSD, Apache, GPL, etc, is not acceptable. Neither are WTFPL and [Unlicense](https://unlicense.org).
	- Place a file named `license` or `LICENSE` in the repo root with the license text.
	- **Do not** add the license name or text to the readme. GitHub already shows the license name at the top of the repo.
	- To verify that you've read all the guidelines, please comment on your pull request with just the word `unicorn`.
- Has [contribution guidelines](awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't use [hard-wrapping](https://stackoverflow.com/questions/319925/difference-between-hard-wrap-and-soft-wrap).
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Doesn't include an `Inspired by awesome-foo` or `Inspired by the Awesome project` kinda link at the top of the readme. The Awesome badge is enough.

**Go to the top and read it again.**

sindresorhus commented:
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

KeyboardInterrupt commented:
Hi @iamericfletcher,
Thank you for your feedback I addressed most of it:

> There is an open issue to address resources that are no longer maintained.

I resolved the Issue and actually had to remove a few things from the list because of them no longer being maintained, and I Updated Other parts. [Link.](https://github.com/KeyboardInterrupt/awesome-ansible/pull/33)

> This pull request does not comply with list requirements.
> _This pull request has a title in the format Add Name of List - add Ansible should be Add Awesome-Ansible._

Fixed it :)

> Perhaps a more detailed description in the About section?

I think it is descriptive of the lists content. :thinking:  I cannot come up with something clever right now, but I will give it some more thought!.

iamericfletcher commented:
Good afternoon @KeyboardInterrupt. Thanks for the update. The list looks good from my perspective. However, this pull request is missing something. Please review the [list guidelines](https://github.com/sindresorhus/awesome/blob/main/pull_request_template.md) line by line.

KeyboardInterrupt commented:
I fixed a typo in the Description, apart from that, I have studied the Guidelines a few more times now and I cannot find anything else that might be a Problem.

I am Sorry, but do you mind pointing me in the right direction a bit more? Maybe I misunderstood something?

iamericfletcher commented:
@KeyboardInterrupt please review the line items under the section

> Has an appropriate license.

KeyboardInterrupt commented:
I am so sorry!
As the license of the List is `CC0-1.0` It did not occur to me that there might linger an Issue within the list items of the license Section. My Eyes must have skimmed over it at least 3 Times... :eyes:!

I made sure to re read the Whole Thing again!

KeyboardInterrupt commented:
unicorn

keller-mark commented:
I believe your list conforms to all guidelines. However for consistency with the rest of your list you may want to add a description to its Videos section as you have done with the other sections. https://github.com/KeyboardInterrupt/awesome-ansible#videos

You may also want to separate the word "Blogposts" into "Blog posts" in this section for more accurate grammar. https://github.com/KeyboardInterrupt/awesome-ansible#blogposts-and-opinions

keller-mark commented on readme.md:
```diff
@@ -695,6 +695,7 @@
 - [Biomedical Information Extraction](https://github.com/caufieldjh/awesome-bioie#readme) - How to extract information from unstructured biomedical data and text.
 - [Web Archiving](https://github.com/iipc/awesome-web-archiving#readme) - An effort to preserve the Web for future generations.
 - [WP-CLI](https://github.com/schlessera/awesome-wp-cli#readme) - Command-line interface for WordPress.
+- [Ansible](https://github.com/KeyboardInterrupt/awesome-ansible#readme) - Ansible is a Python-based, open source IT configuration management (CM) and automation platform.
```
```suggestion
- [Ansible](https://github.com/KeyboardInterrupt/awesome-ansible#readme) - A Python-based, open source IT configuration management (CM) and automation platform.
```

The word Ansible may be redundant here.

KeyboardInterrupt commented:
@keller-mark Thank you for your Feedback, I applied your suggested changes. ;)

axsaucedo commented:
PR looks good to go imo

sindresorhus commented:
The PR title is incorrect.

sindresorhus commented:
Linkify `Ansible` in `Ansible is an open source toolkit, written in Python` to its website.

sindresorhus commented:
> The image should link to the project website or any relevant website.

sindresorhus commented:
Entry descriptions should not be in title-case. Same with the section descriptions.

sindresorhus commented:
> Ansible is an open source toolkit, written in Python, it is used for Configuration Management, Application Deployment, Continuous Delivery, IT infrastructure automation and automation in general.

This should not be titled-cased.

KeyboardInterrupt commented:
Hi, Thank you for your Feedback.

> Linkify `Ansible` in `Ansible is an open source toolkit, written in Python` to its website.
> >The image should link to the project website or any relevant website.

When applying this awesome-linter started complaining that I [had a double link](https://github.com/KeyboardInterrupt/awesome-ansible/runs/1199283821#step:3:9).
I worked around this issue by adding `<!--lint disable double-link-->` and `<!--lint enable double-link-->` comments.
I hope this is okay, and that I should not have resolved this in another way.

Regarding the casing, this should be fixed now. Some words are still Cased, e.g. Role and Playbook as those are also cased in the Official Documentation. Thank you for pointing out these grammatical mistakes, as English is not my native language. I will try to find a better spellchecker ;)

sindresorhus commented:
Tweet: https://twitter.com/awesome__re/status/1323063386687627265
